Blahgger/Controllers/AccountController.cs

```diff
@@ -0,0 +1,120 @@
+﻿using Blahgger.Models;
+using System.Configuration;
+using System.Data.SqlClient;
+using System.Web.Mvc;
+using System.Web.Security;
+
+namespace Blahgger.Controllers
+{
+    public class AccountController : Controller
+    {
+        [AllowAnonymous]
+        public ActionResult Login()
+        {
+            // string hashedPassword = BCrypt.Net.BCrypt.HashPassword("password", 12);
+
+            return View();
+        }
+
+        [AllowAnonymous]
+        public ActionResult Register()
+        {
+            return View();
+        }
+
+        [HttpPost]
+        public ActionResult Login(LoginUser user)
+        {
+            // Did the user provide values?
+            if (ModelState.IsValidField("Email") && ModelState.IsValidField("Password"))
+            {
+                // TODO get the user record from DB.
+                string connectionString = ConfigurationManager.ConnectionStrings["BlahggerDatabase"].ConnectionString;
+
+                using (SqlConnection connection = new SqlConnection(connectionString))
+                {
+                    connection.Open();
+                    string sql = @"
+                            SELECT HashedPassword
+                            FROM [User]
+                            WHERE Email = @Email
+                        ";
+                    SqlCommand command = new SqlCommand(sql, connection);
+                    command.Parameters.AddWithValue("@Email", user.Email);
+                    SqlDataReader reader = command.ExecuteReader();
+
+                    string hashedPassword = null;
+
+                    while (reader.Read())
+                    {
+                        hashedPassword = reader.GetString(0);
+                    }
+
+                    // Check if the password matches.
+                    if (string.IsNullOrEmpty(hashedPassword) || 
+                        !BCrypt.Net.BCrypt.Verify(user.Password, hashedPassword))
+                    {
+                        // Failed to find a record with the provided email.
+                        ModelState.AddModelError("", "Login failed.");
+                    }
+                }
```

Comment:
Nice job extending my example to include the interaction with the database and password hashing.